src/runs.rs

```diff
@@ -305,9 +318,28 @@ impl DaemonRun {
         let system = SystemInfo::get();
 
         match system.process(pid) {
-            Some(process) => match process.kill() {
-                true => Ok(()),
-                false => miette::bail!("Failed to kill process with pid '{}'.", pid.as_u32()),
+            // First try to terminate the process with a SIGTERM signal
+            Some(process) => match process.kill_with(sysinfo::Signal::Term) {
```

Comment:
I just tried with your latest branch and it still doesn't kill jupyterlab for me. Maybe we need to send it to the child PID or something?

Reply:
I am done with the rest and will focus on kill now.

I tried quick with TERM and indeed it does not work (neither the `examples/daemon/` by the way). But I wasn't expecting this to work.

I will try retrieving the child processes.

Reply:
I think I got it to work. It's simply about iterating over all the children and killing them as well.

It requires iterating over all the processes of the system, but it's working for both `daemon` and `jupyterlab`.

Reply:
You can try with the latest commit of this branch.